lib/segment/src/vector_storage/simple_sparse_vector_storage.rs

```diff
@@ -162,8 +163,14 @@ impl VectorStorage for SimpleSparseVectorStorage {
         self.total_vector_count
     }
 
-    fn size_in_bytes(&self) -> usize {
-        self.total_sparse_size * (std::mem::size_of::<f32>() + std::mem::size_of::<u32>())
+    fn size_of_available_vectors_in_bytes(&self) -> usize {
+        if self.total_vector_count == 0 {
+            return 0;
+        }
+        let available_fraction =
+            (self.total_vector_count - self.deleted_count) as f32 / self.total_vector_count as f32;
+        let available_size = (self.total_sparse_size as f32 * available_fraction) as usize;
```

Comment:
How about f64 here as this will only accurately represent counts up to 16.7 million. Or do we not care from that point onwards?

Reply:
On second thought, I don't expect any real problems here.